vision/Object Detection/standard_obj_text.py

```diff
@@ -2,76 +2,116 @@
 Algorithms related to detecting the text of standard objects.
 """
 
+from typing import Dict, List, Tuple, Optional
+
 import cv2
+
 import numpy as np
+import numpy.typing as npt
+
 import pytesseract
 
+# Possible colors and HSV upper/lower bounds
+POSSIBLE_COLORS: Dict[str, npt.NDArray[np.int64]] = {
+    "WHITE": np.array([[180, 18, 255], [0, 0, 231]]),
+    "BLACK": np.array([[180, 255, 30], [0, 0, 0]]),
+    "GRAY": np.array([[180, 18, 230], [0, 0, 40]]),
+    "RED": np.array(
+        [[180, 255, 255], [159, 50, 70], [9, 255, 255], [0, 50, 70]]
+    ),  # red wraps around and needs 2 ranges
+    "BLUE": np.array([[128, 255, 255], [90, 50, 70]]),
+    "GREEN": np.array([[89, 255, 255], [36, 50, 70]]),
+    "YELLOW": np.array([[35, 255, 255], [25, 50, 70]]),
+    "PURPLE": np.array([[158, 255, 255], [129, 50, 70]]),
+    "BROWN": np.array([[20, 255, 180], [10, 100, 120]]),
+    "ORANGE": np.array([[24, 255, 255], [10, 50, 70]]),
+}
+
 
 class TextCharacteristics:
     """
     Class for detecting characteristics of text on standard objects.
-    Characteristics consist of the character, orientation, and color.
+    Characteristics are of the character, orientation, and color.
     """
 
-    def __init__(self):
-        self.image = np.array([])
+    def __init__(self) -> None:
+        # related to text detection
+        self.rotated_img: npt.NDArray[np.uint8] = np.array([])
+        self.preprocessed: npt.NDArray[np.uint8] = np.array([])
 
-    def get_text_characteristics(self, img: np.ndarray, bounds: np.ndarray) -> tuple:
+        # related to text color
+        self.text_cropped_img: npt.NDArray[np.uint8] = np.array([])
+        self.kmeans_img: npt.NDArray[np.uint8] = np.array([])
+        self.color: Optional[str] = None
+
+        # related to text orientation
+        self.orientation: Optional[str] = None
+
+    def get_text_characteristics(
+        self, img: npt.NDArray[np.uint8], bounds: List[List[int]]
+    ) -> Tuple[Optional[str], Optional[str], Optional[str]]:
         """
         Gets the characteristics of the text on the standard object.
 
         Parameters
         ----------
-        img : np.ndarray
+        img : np.NDArray[np.uint8]
             image to find characteristics of text within
-        bounds : np.ndarray
+        bounds : List[List[int, int]]
             bounds of standard object containing text on in image
 
         Returns
         -------
-        tuple
-            characteristics of the text in the form (character, orientation, color)
+        (character, orientation, color) : Tuple[str, str, str]
```

Comment:
This is not just a tuple of strings, but a tuple of Optional strings, so you should label it as such (either with `Optional[str]` or `str | None` for each str in the tuple, describing what the meaning of a None type output means for each value of the tuple